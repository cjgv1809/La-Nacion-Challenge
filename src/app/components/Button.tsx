interface ButtonProps {
  text: string;
}

function Button({ text }: ButtonProps) {
  return (
    <div className="col-12 hlp-text-center hlp-margintop-40">
      <button type="button" className="--btn --secondary">
        {text}
      </button>
    </div>
  );
}

export default Button;
