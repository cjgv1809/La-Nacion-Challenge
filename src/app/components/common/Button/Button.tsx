interface ButtonProps {
  text: string;
  className?: string;
  onClick?: () => void;
}

function Button({
  text,
  className = "--btn --secondary",
  onClick,
}: ButtonProps) {
  return (
    <button type="button" className={className} onClick={onClick}>
      {text}
    </button>
  );
}

export default Button;
