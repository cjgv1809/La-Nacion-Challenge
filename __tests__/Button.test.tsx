import { render, screen } from "@testing-library/react";
import Button from "@/app/components/common/Button";

describe("Button", () => {
  it("renders a button", () => {
    render(<Button text="Click me" />);

    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
  });

  it("renders the exact text passed to it", () => {
    render(<Button text="Click me" />);

    const button = screen.getByRole("button");

    expect(button).toHaveTextContent("Click me");
  });

  it("renders the correct class", () => {
    render(<Button text="Click me" />);

    const button = screen.getByRole("button");

    expect(button).toHaveClass("--btn --secondary");
  });

  it("the text passed to it is not empty", () => {
    render(<Button text="Click me" />);

    const button = screen.getByRole("button");

    expect(button).not.toHaveTextContent("");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Button text="Click me" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
