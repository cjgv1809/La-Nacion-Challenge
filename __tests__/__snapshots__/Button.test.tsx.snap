// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="--btn --secondary"
    type="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
